_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,669 @@
+
+Run the following command to create the AKS cluster.
+
+`az aks create --resource-group australia-east --name jhipster-demo --node-count 2 --enable-addons monitoring --generate-ssh-keys --node-vm-size standard_b4ms`
+
+This will probably take a few minutes.
+
+As a side note, at any point, you can stop the cluster. This will pause billing on the cluster.
+
+```bash
+az aks stop --resource-group australia-east --name jhipster-demo 
+```
+
+And you can start it again.
+
+```bash
+az aks start --resource-group australia-east --name jhipster-demo 
+```
+
+You can also stop and start the cluster from the Azure portal as well. 
+
+The next step is to get the credentials for the cluster and merge them into `.kube/confg` so that `kubectl` can use them. Use the following command.
+
+```bash
+az aks get-credentials --resource-group australia-east --name jhipster-demo
+```
+
+You should see output that ends like the following.
+
+```bash
+Merged "jhipster-demo" as current context in /home/andrewcarterhughes/.kube/config
+```
+
+You should be able to use `kubectl` to get the node on Azure.
+
+```bash
+kubectl get nodes
+```
+
+```bash
+NAME                                STATUS   ROLES   AGE   VERSION
+aks-nodepool1-58487107-vmss000000   Ready    agent   12m   v1.22.6
+```
+
+You can also  list a lot of information about the cluster in JSON format using:
+
+```bash
+az aks list --resource-group australia-east
+```
+
+#### Configure Kubernetes for Okta OAuth and Cosmos DB
+
+The Kubernetes files in the `k8s` directory were created with the JHipster Kubernetes sub-generator ([see the docs for info](https://www.jhipster.tech/kubernetes/)). To see how the original project was generated, take a look at [Matt Raible's tutorial](https://developer.okta.com/blog/2021/06/01/kubernetes-spring-boot-jhipster). As outlined above, these files were modified to work with Azure Cosmos DB instead of MongoDB in a Kubernetes pod (which is what the sub-generator assumes).
+
+Configure Spring OAuth in the Kubernetes pod by updating `k8s/registry-k8s/application-configmap.yml`. You can use the same values you used above in the Docker Compose section, in `docker-compose/central-server-config/application.yml`.
+
+Make sure you use the encrypted client secret enclosed in quotes with the `{cipher}` prefix. You're going to copy both the encryption key and the encrypted client secret from the Docker Compose configuration to avoid having to re-encrypt the client secret with a new key.
+
+```
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: "{cipher}<encrypted-client-secret>"
```

Comment:
This doesn't match the start tag - there's no `{cipher}` in this file. Not a big deal, but we might want to match them match too.